### PR TITLE
feat: support new username system

### PR DESCRIPTION
**Pull request by Snazzah**
Ref: https://github.com/discord/discord-api-docs/pull/6130

### CODE REVIEW

File: src/constants.ts [hunk 1]
```diff
@@ -581,6 +581,7 @@ export interface CommandUser {
   username: string;
   avatar: string | null;
   avatar_decoration: string | null;
+  global_name: string | null;
   discriminator: string;
```
Comment (erkinalp) [inline src/constants.ts:585]: for further future proofing (once everyone gets unique usernames),
```suggestion
  discriminator?: string;
```

File: src/structures/user.ts [hunk 2]
```diff
@@ -57,6 +60,7 @@ export class User {
 
   /** The hash for the default avatar of a user if there is no avatar set. */
   get defaultAvatar() {
+    if (this.discriminator === '0') return Number((BigInt(this.id) << 22n) % 5n);
```
Comment (NickantX) [inline src/structures/user.ts:63]:
```suggestion
    if (this.discriminator === '0') return Number((BigInt(this.id) >> 22n) % 5n);
```

User ID should be right-shifted, not left-shifted.

**Comment by erkinalp** [comment]
Should be merged, given some users currently already have new usernames.

File: src/constants.ts (hunk 1 above)
Comment (Snazzah) [reply src/constants.ts:585]: Would rather match closer to the PR's documentation. Discriminators are not optional.

**Pull request merged**